src/main/java/lotto/domain/generator/LottoAutoGenerator.java

```diff
@@ -0,0 +1,44 @@
+package lotto.domain.generator;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import lotto.domain.lotto.Number;
+import lotto.domain.lotto.Numbers;
+
+public class LottoAutoGenerator implements LottoGenerator {
+
+    private List<Integer> numbers;
+
+    public LottoAutoGenerator() {
+        generate();
+        shuffle();
+    }
+
+    private void shuffle() {
+        Collections.shuffle(numbers);
+    }
+
+    private void generate() {
+        numbers = IntStream.rangeClosed(START_NUMBER, END_NUMBER)
+            .boxed()
+            .collect(Collectors.toList());
+    }
+
+    @Override
+    public Numbers generateNumbers() {
+        List<Integer> numbers = new ArrayList<>();
+        for (int i = 0; i < LOTTO_NUMBERS; i++) {
+            numbers.add(this.numbers.get(i));
+        }
+
+        return new Numbers(
+            numbers.stream()
+                .sorted()
+                .map(Number::new)
+                .collect(Collectors.toList())
+        );
+    }
```

Comment:
```suggestion
    private List<Number> lottoNumberPool;

    public LottoAutoGenerator() {
        this.lottoNumberPool = IntStream.rangeClosed(LOTTO_MIN_NUMBER, LOTTO_MAX_NUMBER)
          .mapToObj(LottoNumber::new)
          .collect(Collectors.toList());
    }

    @Override
    public LottoNumbers generateNumbers() {
        Collections.shuffle(lottoNumberPool);
        return new LottoNumbers(lottoNumberPool.stream()
          .limit(LOTTO_NUMBERS)
          .collect(Collectors.toList()));
    }
```
로또 번호 풀을 만들고 셔플 한 뒤에 6개를 잘라 티켓을 만드는 로직은 같습니다.

로또 숫자를 Integer로 표현하는 것보다 Number로 표현하는 것은 어떨까요?
상수/객체의 의미를 명확하게 하기 위해 네이밍을 수정해 보는 것은 어떨까요?

Reply:
훨씬 코드가 명확하게 파악되는 것 같습니다. 
네이밍을 좀 더 명확하게 할 수 있도록 많이 생각하겠습니다!
감사합니다.